### PR TITLE
Extract provision

### PR DESCRIPTION
**Pull request by cweibel**

### CODE REVIEW

File: provision.sh [hunk 1]
```diff
@@ -159,30 +177,46 @@ rm spiff_linux_amd64.zip
...
 # Upload the bosh release, set the deployment, and execute
-bosh upload release https://bosh.io/d/github.com/cloudfoundry/cf-release?v=${cfReleaseVersion}
-bosh deployment cf-openstack-${CF_SIZE}
-bosh prepare deployment
+deployedVersion=$(bosh releases | grep " ${cfReleaseVersion}" | awk '{print $4}')
+deployedVersion="${deployedVersion//[^[:alnum:]]/}"
+if [[ ! "$deployedVersion" == "${cfReleaseVersion}" ]]; then
+  bosh upload release https://bosh.io/d/github.com/cloudfoundry/cf-release?v=${cfReleaseVersion}
+  bosh deployment cf-openstack-${CF_SIZE}
+  bosh prepare deployment || bosh prepare deployment  #Seems to always fail on the first run...
```
Comment (rkoster) [inline provision.sh:185]: what is the error you are getting?

File: provision/prepare-provision [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+fail() {
+  echo "$*" >&2
+  exit 1
+}
+
+cp -R provision.sh provision/provision.sh
+(( "$?" == "0" )) ||
+  fail "Could not copy the original provision.sh file to the provision folder."
+
+# TERRAFORM_DIR=${TERRAFORM_DIR:-terraform-aws-cf-install}
```
Comment (lnguyen) [inline provision/prepare-provision]: whats up with all this commented out stuff?

File: provision.sh (hunk 1 above)
Comment (cweibel) [inline provision.sh:185]: It's not to detect an error, it's for if you run the script a second time to skip uploading the release if that version of the release has already been successfully uploaded.

File: provision/prepare-provision (hunk 2 above)
Comment (cweibel) [inline provision/prepare-provision]: Pushing a new version without the commented out lines which weren't needed.

File: provision.sh (hunk 1 above)
Comment (rkoster) [inline provision.sh:185]: I'm referring to `#Seems to always fail on the first run...`, also bosh prepare deployment is idempotent.

**Pull request merged**